### PR TITLE
added authentication : OAuth 2.0

### PR DESCRIPTION
**Pull request by Abhinav200207**
### Fixes #<number>
<Do not delete this template>
Description: <provide a description> The title should be short and relevant.

#### Checklist
- [x] I've followed the Contributing guidelines provided in the repository.
- [x] I've made the changes which were demanded in the linked issue.
- [x] I've tested my code on a Chromium based browser.
- [ ] I've tested my code on Mozilla Firefox.
- [x] My code gave a clean console on debugging. (no warnings/errors)

#### Testimonial
<provide a relevant screenshot or any testimonial for the solved issue if applicable>

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -23,8 +26,12 @@ app.use(express.json());
 app.use(express.urlencoded({ extended: false }));
 app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
+app.use(session({ secret: 'cats', resave: false, saveUninitialized: true }));
```
Comment (EmperorYP7) [inline server/app.js:29]:
```suggestion
app.use(session({ secret: process.env.SESSION_SECRET, resave: false, saveUninitialized: true }));
```

**Comment by SrijanSriv** [comment]
We would require the heroku logs to render index properly. Discussing the same in the next meet.

File: server/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+.env
```
Comment (EmperorYP7) [inline server/.gitignore:1]:
```suggestion
.env
!.env.dev
```

**Review by EmperorYP7 — CHANGES_REQUESTED** [review]
A few changes in the code are needed. Testing is yet to be done.

**Pull request closed without merging**